### ISSUE TITLE
Error in if (nMinor < 0) { : missing value where TRUE/FALSE needed Calls: callSubclones -> determine_copynumber

### ISSUE THREAD

**Issue by TakaYamaguchi** [Wedge-lab/battenberg#37]
Hi,

This is very rare but a few samples had this issue below when calling callSubclones().

Error in if (nMinor < 0) { : missing value where TRUE/FALSE needed
Calls: callSubclones -> determine_copynumber

This may be due to ploidy NA  (FRAC_GENOME) in sample_rho_and_psi.txt?

rho          psi   ploidy  distance          is.best
ASCAT        1     2.15    2.10143838105564  NA       NA
FRAC_GENOME  0.95  **NA**      2.09906082321258  0        FALSE
REF_SEG      1     2       2                 1        TRUE

Thanks,

Taka

**Comment by Sames-Jtudd** [comment]
I have the same error. The error orginated in the follwing line of the command Battenberg:::determine_copynumber
33    nMinor = (rho - 1 + (1 - l) * psi * 2^(LogR/gamma))/rho

the reason was that l was "NA". l is derived from the BAFsegment file segment column, which were all NA in my case. Not sure why.  If you figure it out let me know.

**Comment by keiranmraine** [comment]
We see this frequently and abandon the analysis of these samples.  It would be useful if there was an appropriate error message indicating the cause of this.

**Comment by lydiayliu** [comment]
I also ran into this issue with our `bwa-mem2` aligned samples that are of East Asian ancestry to GRCh38.

After following the line of inquiry given above, I confirmed that the error is due to `l` being `NA` in
```
nMinor = (rho - 1 + (1 - l) * psi * 2^(LogR/gamma))/rho
```
in `Battenberg:::determine_copynumber`.
This is due to the `.BAFsegment.` file containing `NA` for stretches of multiple chromosomes. Following this problem upstream I realized that in the `_alleleFrequencies_chr` files, the problematic regions have `good depth = 0`.

Further investigation revealed that the reason there is no depth in those regions is because the regions have ALT CONTIGS where all the reads were aligned to instead, resulting in `depth = 0` in the `alleleCounter` result for the regions. To properly fix this issue, one probably needs to align with a reference genome where the [ALT CONTIGS are removed](https://lh3.github.io/2017/11/13/which-human-reference-genome-to-use).

I'm also trying to avoid the problematic regions (since they are very small anyways) by adding a check to skip the segment if `nMinor` is `NA`...
